text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,1443 @@
+survey we did on how people evaluate crates by hand today.
+
+A few assumptions we made:
+
+- Measures that can be made automatically are preferred over measures that
+  would need administrators, curators, or the community to spend time on
+  manually.
+- Measures that can be made for any crate regardless of that crate's choice of
+  version control, repository host, or CI service are preferred over measures
+  that would only be available or would be more easily available with git,
+  GitHub, Travis, and Appveyor. Our thinking is that when this additional
+  information is available, it would be better to display a badge indicating it
+  since this is valuable information, but it should not influence the ranking
+  of the crates.
+- There are some measures, like "suitability for the current task" or "whether
+  I like the way the crate is implemented" that crates.io shouldn't even
+  attempt to assess, since those could potentially differ across situations for
+  the same person looking for a crate.
+- We assume we will be able to calculate these in a reasonable amount of time
+  either on-demand or by a background job initiated on crate publish and saved
+  in the database as appropriate. We think the measures we have proposed can be
+  done without impacting the performance of either publishing or browsing
+  crates noticeably. If this does not turn out to be the case, we will have to
+  adjust the formula.
+
+## Factors
+
+Through [the survey we conducted][user-research], we found that when people
+evaluate crates, they are looking primarily for approximate signals of:
+
+- Ease of use
+- Maintenance
+- Quality
+
+Feeding those signals are related measures of:
+
+- Popularity
+- Credibility
+
+We detail how we propose to address each of these in turn, plus a rating of the
+five crates from the user research survey as examples.
+
+We'd like to provide a coarse binning of the scores in each category, to avoid
+over-analyzing the difference between, say, 72% and 78% and seeing significance
+where there isn't really one. We've considered using letter grades, but those
+often have emotional associations (F means you're a failure), when it should be
+just an indicator of reality and not a value judgment. So we're also proposing
+an option of an emoji scale and are open to other proposals:
+
+<table>
+    <tr>
+        <th>Percentage</th>
+        <th>Letter grade</th>
+        <th>Emoji</th>
+    </tr>
+    <tr>
+        <td>>= 90%</td>
+        <td>A</td>
+        <td>☀️</td>
+    </tr>
+    <tr>
+        <td>80-89%</td>
+        <td>B</td>
+        <td>🌤</td>
+    </tr>
+    <tr>
+        <td>70-79%</td>
+        <td>C</td>
+        <td>⛅️</td>
+    </tr>
+    <tr>
+        <td>60-69%</td>
+        <td>D</td>
+        <td>🌥</td>
+    </tr>
+    <tr>
+        <td>50-59%</td>
+        <td>E</td>
+        <td>☁️</td>
+    </tr>
+    <tr>
+        <td>&lt;= 49%</td>
+        <td>F</td>
+        <td>🌧</td>
+    </tr>
+</table>
```

Comment:
@nagisa 

> instead of binning crates into buckets, just explicitly note qualities a crate has: “popular parsing crate”, “credible serialization crate”, “easy to use high performance computing crate” etc. While not every crate would receive a nice label, you’d achieve similar results and avoid issues with developers of crates which do not make the cut.

Maybe instead of a third party giving a crate a label which would most likely be argued ad infinitum, crate owners/creators themselves could be given the option to tag their crate as something like; "Production ready" or "work in progress, intended for production" or "not intended for production". So if you searched for serialization crate you'd only get hits like Serde and not something that was never intended to be used by anyone else. Better to completely forget about the abandoned, unusable ones than have them clog up the search results where the majority have a tag of "not unusable" or "abandoned" and only ones tagged for production get reviewed for quality.